packages/bpk-component-dialog/src/BpkDialog.tsx

```diff
@@ -53,31 +55,44 @@ const BpkDialog = ({
     );
   }
 
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && dismissible && onClose) {
+        onClose();
+      }
+    };
+
+    if (isOpen) {
+      window.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen, onClose, dismissible]);
+
   return (
-    <Portal
-      isOpen={isOpen}
-      onClose={onClose}
-      renderTarget={renderTarget}
-      closeOnEscPressed={dismissible}
-    >
-      <BpkDialogInner
-        onClose={onClose}
-        closeOnScrimClick={dismissible}
-        containerClassName={getClassName('bpk-dialog__container')}
-        contentClassName={
-          headerIcon ? getClassName('bpk-dialog--with-icon') : undefined
-        }
-        {...rest}
-      >
-        {headerIcon && <div className={headerIconClassNames}>{headerIcon}</div>}
-        {dismissible && (
-          <span className={closeButtonClassNames}>
-            <BpkCloseButton label={closeLabel} onClick={onClose} />
-          </span>
-        )}
-        {children}
-      </BpkDialogInner>
-    </Portal>
+    <>
+      {isOpen && (
+        <BpkDialogInner
+          onClose={onClose}
+          closeOnScrimClick={dismissible}
```

Comment:
I'm a little confused because actually I can't find this prop in `BpkDialogInner` 🤔 I see we were already passing it, but I wonder if it actually has an effect?

Reply:
This goes through to the `withScrim` wrapper that is around the Inner, this was part of the original API and not been touched so would remain as it was :) 

https://github.com/Skyscanner/backpack/blob/main/packages/bpk-scrim-utils/src/withScrim.tsx#L66

All that has happened here is the `isOpen` wrapper :)

Reply:
Makes sense, yeah, somehow my eyes totally skipped over the export so I just didn't see the scrim 😂